### PR TITLE
Change height schematic Island

### PR DESCRIPTION
**Pull request by DoctaEnkoda**
#399

### CODE REVIEW

File: src/main/java/com/iridium/iridiumskyblock/managers/SchematicManager.java [hunk 1]
```diff
@@ -76,21 +77,21 @@ public SchematicManager(@NotNull File parent) {
...
+        pasteSchematic(island, world, schematic, schematicWorld.yHeightIsland, completableFuture, 0, delay);
 
         return completableFuture;
     }
 
     /**
      * Pastes the schematic one layer at a time starting at the specified y level
-     *
-     * @param island            The island the schematic is being pasted at
+     *  @param island            The island the schematic is being pasted at
```
Comment (MertUnverdi) [inline src/main/java/com/iridium/iridiumskyblock/managers/SchematicManager.java:87]:
```suggestion
     * @param island            The island the schematic is being pasted at
```

File: src/main/java/com/iridium/iridiumskyblock/managers/SchematicManager.java [hunk 2]
```diff
@@ -100,17 +101,17 @@ private void pasteSchematic(final Island island, final World world, final Schema
         // Loop all blocks in the schematics layer at the current y level
         for (int x = 0; x < schematic.getLength(); x++) {
             for (int z = 0; z < schematic.getWidth(); z++) {
-                Block block = island.getCenter(world).subtract(schematic.getLength()/2.00, -90, schematic.getWidth()/2.00).add(x, y, z).getBlock();
+                Block block = island.getCenter(world).subtract(schematic.getLength()/2.00, -90, schematic.getWidth()/2.00).add(x, (-90 + yHeightIsland) + y, z).getBlock();
```
Comment (MertUnverdi) [inline src/main/java/com/iridium/iridiumskyblock/managers/SchematicManager.java:104]: why doing here `(-90 + yHeightIsland)` im didn't understand could you explain it, why not directly yHeightIsland ?

**Comment by MertUnverdi** [comment]
i've viewed it and everything seems right, verified pr 👍🏻

File: src/main/java/com/iridium/iridiumskyblock/managers/SchematicManager.java (hunk 2 above)
Comment (DoctaEnkoda) [reply src/main/java/com/iridium/iridiumskyblock/managers/SchematicManager.java:104]: Basically it is created at the height of 90 blocks, we subtract the 90 to return to 0 base and we let the server choose

Comment (MertUnverdi) [reply src/main/java/com/iridium/iridiumskyblock/managers/SchematicManager.java:104]: Okay so just do it as `yHeightIsland - 90` :panda:

Comment (MertUnverdi) [reply src/main/java/com/iridium/iridiumskyblock/managers/SchematicManager.java:104]: Or directly add it to subracting part will be more better

Comment (MertUnverdi) [reply src/main/java/com/iridium/iridiumskyblock/managers/SchematicManager.java:104]:
```suggestion
                Block block = island.getCenter(world).subtract(schematic.getLength()/2.00, 0, schematic.getWidth()/2.00).add(x,  yHeightIsland + y, z).getBlock();
```

File: src/main/java/com/iridium/iridiumskyblock/configs/Schematics.java [hunk 3]
```diff
@@ -62,10 +62,12 @@
     public static class SchematicWorld {
         public XBiome biome;
         public String schematicID;
+        public double yHeightIsland;
 
-        public SchematicWorld(XBiome biome, String schematicID) {
+        public SchematicWorld(XBiome biome, String schematicID, double yheightisland) {
```
Comment (dlsf) [inline src/main/java/com/iridium/iridiumskyblock/configs/Schematics.java:67]: Two issues:
1. Please user lowerCamelCase for variable names
2. I think the name `islandHeight` would be a better name. Height already implies that the y axis is meant.

File: src/main/java/com/iridium/iridiumskyblock/managers/SchematicManager.java [hunk 4]
```diff
@@ -100,17 +101,17 @@ private void pasteSchematic(final Island island, final World world, final Schema
         // Loop all blocks in the schematics layer at the current y level
         for (int x = 0; x < schematic.getLength(); x++) {
             for (int z = 0; z < schematic.getWidth(); z++) {
-                Block block = island.getCenter(world).subtract(schematic.getLength()/2.00, -90, schematic.getWidth()/2.00).add(x, y, z).getBlock();
+                Block block = island.getCenter(world).subtract(schematic.getLength()/2.00, 0, schematic.getWidth()/2.00).add(x, yHeightIsland + y, z).getBlock();
```
Comment (dlsf) [inline src/main/java/com/iridium/iridiumskyblock/managers/SchematicManager.java:104]: Not an issue with your code but let's change it :p
```suggestion
                Block block = island.getCenter(world).subtract(schematic.getLength() / 2.0, 0, schematic.getWidth() / 2.0).add(x, yHeightIsland + y, z).getBlock();
```

**Review by dlsf — CHANGES_REQUESTED** [review]

File: src/main/java/com/iridium/iridiumskyblock/configs/Schematics.java [hunk 5]
```diff
@@ -62,10 +62,12 @@
     public static class SchematicWorld {
         public XBiome biome;
         public String schematicID;
+        public double yHeightIsland;
```
Comment (dlsf) [inline src/main/java/com/iridium/iridiumskyblock/configs/Schematics.java:65]: In my opinion, this field should be renamed to `islandHeight` too

**Pull request merged**